### PR TITLE
[RFC] Handle account.application.deauthorized event

### PR DESCRIPTION
**Pull request by ticosax**
Not sure what is best ? deleting the account ? or just mark it.
- deleting is kind of bad for accountability
- but keeping it will make re-association a bit more complex

Tha'ts why I think a marker is a better approach

### CODE REVIEW

File: pinax/stripe/tests/test_models.py [hunk 1]
```diff
@@ -163,13 +163,13 @@ def test_subscription_delete(self):
...
-        self.assertEquals(repr(a), "Account(pk=None, display_name='', type=None, stripe_id='')")
+        self.assertEquals(repr(a), "Account(pk=None, display_name='', type=None, stripe_id='', authorized=True)")
         a.stripe_id = "acct_X"
         self.assertEquals(str(a), " - acct_X")
-        self.assertEquals(repr(a), "Account(pk=None, display_name='', type=None, stripe_id='acct_X')")
+        self.assertEquals(repr(a), "Account(pk=None, display_name='', type=None, stripe_id='acct_X', authorized=True)")
         a.display_name = "Display name"
         self.assertEquals(str(a), "Display name - acct_X")
-        self.assertEquals(repr(a), "Account(pk=None, display_name='Display name', type=None, stripe_id='acct_X')")
+        self.assertEquals(repr(a), "Account(pk=None, display_name='Display name', type=None, stripe_id='acct_X', authorized=True)")
```
Comment (blueyed) [inline pinax/stripe/tests/test_models.py]: could use `authorized = False` in one of them (the last one), but minor.

File: pinax/stripe/tests/test_webhooks.py [hunk 2]
```diff
@@ -183,29 +183,29 @@ def signal_handler(sender, *args, **kwargs):
...
         self.assertTrue(EventProcessingException.objects.filter(event=event).exists())
 
     @patch("pinax.stripe.actions.customers.link_customer")
     @patch("pinax.stripe.webhooks.Webhook.validate")
     def test_process_return_none(self, ValidateMock, LinkMock):
         # note: we choose an event type for which we do no processing
-        event = Event.objects.create(kind="account.application.deauthorized", webhook_message={}, valid=True, processed=False)
-        self.assertIsNone(AccountApplicationDeauthorizeWebhook(event).process())
+        event = Event.objects.create(kind="account.external_account.created", webhook_message={}, valid=True, processed=False)
+        self.assertIsNone(AccountExternalAccountCreatedWebhook(event).process())
```
Comment (blueyed) [inline pinax/stripe/tests/test_webhooks.py:212]: Why were the existing tests changed?
Seems to be better to keep them, but add new ones?!

**Comment by blueyed** [comment]
The marker/field makes sense to me.

File: pinax/stripe/tests/test_webhooks.py (hunk 2 above)
Comment (ticosax) [reply pinax/stripe/tests/test_webhooks.py:212]: per comment:
> note: we choose an event type for which we do no processing

File: pinax/stripe/migrations/0019_account.authorized.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        ('pinax_stripe', '0018_add_discounts'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='account',
+            name='authorized',
+            field=models.BooleanField(default=True),
+        ),
+    ]
```
Comment (blueyed) [inline pinax/stripe/migrations/0019_account.authorized.py]: You cannot have a dot in the file basename: `ModuleNotFoundError: No module named 'pinax.stripe.migrations.0019_account'`

Comment (ticosax) [reply pinax/stripe/migrations/0019_account.authorized.py]: thx

**Comment by ticosax** [comment]
@blueyed please restart the jobs. I can't.

**Comment by ticosax** [comment]
pleasse discard, now I can.

**Pull request merged**